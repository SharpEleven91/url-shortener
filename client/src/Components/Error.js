import React from "react";
import PropTypes from 'prop-types';
import { Typography, Fade } from "@material-ui/core";

const Error = (props) => {
  return (
    <Fade in={props.checked} timeout={{ enter: 1700, exit: 1700 }}>
      <Typography
        data-testid="error-display"
        color="error"
        component="h3"
        variant="h3"
        gutterBottom
      >
        {" "}
        Invalid URL{" "}
      </Typography>
    </Fade>
  );
}

Error.propTypes = {
  checked: PropTypes.bool,
}

export default Error

