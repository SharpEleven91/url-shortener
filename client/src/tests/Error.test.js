import { mount, configure } from 'enzyme';
import React from 'react';
import expect from 'expect';
import Error from '../Components/Error';
import { render } from 'react-testing-library';

test('Component displays error properly', () => {
    const { container } = render(<Error/>);
    const text = container.firstChild;
    expect(text.textContent).toBe(" Invalid URL ");
})