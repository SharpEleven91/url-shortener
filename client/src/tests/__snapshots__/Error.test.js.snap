// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders properly 1`] = `
<h3
  class="MuiTypography-root-1 MuiTypography-h3-15 MuiTypography-colorError-35 MuiTypography-gutterBottom-28"
  data-testid="error-display"
  style="opacity: 1; -webkit-transition: opacity 1700ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
>
   
  Invalid URL
   
</h3>
`;
